rust-runtime/aws-smithy-runtime-api/Cargo.toml

```diff
@@ -12,6 +12,7 @@ repository = "https://github.com/awslabs/smithy-rs"
 [features]
 default = []
 http-auth = ["dep:zeroize"]
+anonymous-auth = []
```

Comment:
Previously, the anonymous auth stuff was always available. I added some more stuff related to anonymous auth and feature-gated it since we were already doing so for `http-auth`.